pkg/eventfilter/benchmarks/common_benchmark_test.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package benchmarks
+
+import (
+	"context"
+	"testing"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+
+	"knative.dev/eventing/pkg/eventfilter"
+)
+
+type FilterBenchmark struct {
+	name  string
+	arg   interface{}
+	event cloudevents.Event
+}
+
+// Avoid DCE
+var Filter eventfilter.Filter
+var Result eventfilter.FilterResult
```

Comment:
I think we should rename those, the usage it's unreadable, too many filter.

Reply:
@slinkydeveloper WDYT?

Reply:
Which names do you propose?

Reply:
IIUC, these are just to avoid DCE. So the variables are really not designed to be used. I would suggest to use lower case if I want to be picky. Also sth like `filterForNoDCE`.

Reply:
To be 100% you exclude DCE, you need variables to be exported

Reply:
I meant package.types not variable names.

trigger.Filter 
trigger.FilterResult 

wdyt?

Reply:
or filter.Interface and filter.Result

Reply:
And where should this `trigger` module should live? Under `eventing/pkg/trigger`? Not sure, to me `eventfilter`, although it's repeated a couple of times, sounds like a good name (and note, this code is not specific at all to trigger, in fact at some point it was even discussed to implement filters at source level)